iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -2095,6 +2097,55 @@ public List<Partition> getPartitionsByExpr(org.apache.hadoop.hive.ql.metadata.Ta
     }
   }
 
+  @Override
+  public List<Partition> getPartitionsWithFilter(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
```

Comment:
how it's different from `getPartitionsByExpr`, only that we check for specId?

Reply:
Yes, `getPartitionsByExpr` returns partitions based on latest spec only.
I'll change it to use the new method to void unnecessary duplication.

Reply:
👍